### PR TITLE
DS-4497 Bump Rack version to 3.1

### PR DESCRIPTION
**Pull request by ildomar-grings**
## Overview

[DS-4497](https://loyaltynz.atlassian.net/browse/DS-4497)

## What is the change?
- Bump `Rack` version to 3.1
- Drop compatibility with Ruby `2.3.7`
- Rename from `Rack::Handler::AlchemyFlux` to `Rackup::Handler::AlchemyFlux`, to follow `Rack` new gems naming convention
- Regenerated docs

## Notes
Rackup 2.1 requires Ruby >= 2.4
https://rubygems.org/gems/rackup/versions/2.1.0?locale=en

Rack 3.1 requires Ruby >= 2.4
https://rubygems.org/gems/rack

https://github.com/rack/rack/blob/main/UPGRADE-GUIDE.md
Rack 2 & Rack 3 compatibility.

...

bin/rackup, Rack::Server, Rack::Handlerand Rack::Lobster were moved to a separate gem.
Previously, the rackup executable was included with Rack. Because WEBrick is no longer a default gem with Ruby, we had to make a decision: either rack should depend on webrick or we should move that functionality into a separate gem. We chose the latter which will hopefully allow us to innovate more rapidly on the design and implementation of rackup separately from "rack the interface".

In Rack 3, you will need to include:
```
gem 'rackup'
```
This provides all the previously available functionality.

[DS-4133]: https://loyaltynz.atlassian.net/browse/DS-4133?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DS-4497]: https://loyaltynz.atlassian.net/browse/DS-4497?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: doc/AlchemyFlux.html [hunk 1]
```diff
@@ -121,7 +121,7 @@ <h2>
           <div class="docstring">
   <div class="discussion">
     
-<p>Version of the gem, read into &#39;alchemy-flux.gemspec&#39;.</p>
+<p>Version of the gem, read into ‘alchemy-flux.gemspec’.</p>
```
Comment (cloudy9101) [inline doc/AlchemyFlux.html:124]: What's the reason for changes like this?

File: doc/js/app.js [hunk 2]
```diff
@@ -190,15 +191,25 @@ function generateTOC() {
...
+        toc = $('<ol/>');
+        curli.append(toc);
+        curli = undefined;
       }
     }
     if (thisTag < lastTag) {
-      for (i = 0; i < lastTag - thisTag; i++) toc = toc.parent();
+      for (i = 0; i < lastTag - thisTag; i++) {
+        toc = toc.parent();
+        toc = toc.parent();
```
Comment (cloudy9101) [inline doc/js/app.js:206]: ?

**Review by cloudy9101 — APPROVED** [review]
Not sure if those changes in docs are intended.
The upgrade itself looks good to me.

File: doc/js/app.js (hunk 2 above)
Comment (ildomar-grings) [reply doc/js/app.js:206]: Also, this is self-generated by YARD 0.9.36

File: doc/AlchemyFlux.html (hunk 1 above)
Comment (ildomar-grings) [reply doc/AlchemyFlux.html:124]: This is self-generated by YARD 0.9.36

Comment (cloudy9101) [reply doc/AlchemyFlux.html:124]: good to know

**Review by nickpiercy95 — APPROVED** [review]

**Pull request merged**